ecpy_hqc_legacy/tasks/tasks/instr/spdev_tasks.py

```diff
@@ -76,12 +84,23 @@ def check(self, *args, **kwargs):
 
         p1 = locs['freq_1']*locs['duration']*1e-3 if self.ch1_enabled else 1.
         p2 = locs['freq_2']*locs['duration']*1e-3 if self.ch2_enabled else 1.
+        #  THIS CHECK DOES NOT WORK FOR ALAZAR. We need to recalculate the nb
+        #  of samples per trace so that it is multiple of 32. The nb of samples
+        #  is hence modified after this check is performed.
         if (not p1.is_integer() or not p2.is_integer()):
             test = False
             msg = ('The duration must be an integer times the period of the '
                    'demodulations.')
             traceback[self.get_error_path() + '-' + n] = msg
 
+        sampling_rate = self.format_and_eval_string(self.sampling_rate)
+        s1 = sampling_rate % locs['freq_1']*1e6 if self.ch1_enabled else 0
+        s2 = sampling_rate % locs['freq_2']*1e6 if self.ch2_enabled else 0
+        if s1 != 0 or s2 != 0:
+            msg = ('The sampling rate must be a multiple of the demodulation '
+                   'frequency.')
+            traceback[self.get_error_path() + '-' + n] = msg
```

Comment:
If you do not set test to False it means that you consider that this is warning not an error and that the user can easily ignore it. If it is indeed your intent I would suggest to change the message to 'The sampling rate should be...'